### ISSUE TITLE
 average_gradients(model) 

### ISSUE THREAD

**Issue by alicera** [pytorch/tutorials#2305]
## 📚 Documentation

If using DistributedDataParallel as model, does we need to the step of average_gradients(model) ?

```
model = torch.nn.parallel.DistributedDataParallel(model)
```

[DistributedDataParallel](https://pytorch.org/tutorials/intermediate/dist_tuto.html)

```
""" Distributed Synchronous SGD Example """
def run(rank, size):
    torch.manual_seed(1234)
    train_set, bsz = partition_dataset()
    model = Net()
    optimizer = optim.SGD(model.parameters(),
                          lr=0.01, momentum=0.5)

    num_batches = ceil(len(train_set.dataset) / float(bsz))
    for epoch in range(10):
        epoch_loss = 0.0
        for data, target in train_set:
            optimizer.zero_grad()
            output = model(data)
            loss = F.nll_loss(output, target)
            epoch_loss += loss.item()
            loss.backward()
            average_gradients(model)
            optimizer.step()
        print('Rank ', dist.get_rank(), ', epoch ',
              epoch, ': ', epoch_loss / num_batches)


""" Gradient averaging. """
def average_gradients(model):
    size = float(dist.get_world_size())
    for param in model.parameters():
        dist.all_reduce(param.grad.data, op=dist.ReduceOp.SUM)
        param.grad.data /= size
```

cc @mrshenli @osalpekar @H-Huang @kwen2501

**Comment by msaroufim** [comment]
Agreed this is probably not needed, moving to tutorials where this page is hosted

**Comment by subramen** [comment]
No, you don't need to use `average_gradients()` if you're using DDP (`torch.nn.parallel.DistributedDataParallel`) 

Notice that the code snippet doesn't actually use DDP, and therefore requires `average_gradients` to aggregate the gradients from all devices. If you're using DDP, that is taken care of (by the ring all-reduce algorithm). The tutorial this snippet is from illiustrates what is happening under the DDP hood

**Comment by kwen2501** [comment]
Perhaps not directly related to the purpose of this PR, I just want to mention there c10d has average operator:
```
ReduceOp.AVG
```
(so that you don't need to define your own `average_gradients` function)
